bazel/foreign_cc/BUILD

```diff
@@ -1,85 +1,108 @@
 licenses(["notice"])  # Apache 2
 
-load("//bazel:envoy_build_system.bzl", "envoy_package")
-load("@rules_foreign_cc//tools/build_defs:cmake.bzl", "cmake_external")
+load("//bazel:envoy_build_system.bzl", "envoy_cmake_external", "envoy_package")
 
 envoy_package()
 
-cmake_external(
+envoy_cmake_external(
     name = "ares",
     cache_entries = {
         "CARES_SHARED": "no",
         "CARES_STATIC": "on",
         "CMAKE_BUILD_TYPE": "RelWithDebInfo",
     },
-    cmake_options = ["-GNinja"],
     lib_source = "@com_github_c_ares_c_ares//:all",
-    make_commands = [
-        "ninja",
-        "ninja install",
-    ],
-    static_libraries = ["libcares.a"],
+    postfix_script = select({
+        "//bazel:windows_dbg_build": "cp $BUILD_TMPDIR/CMakeFiles/c-ares.dir/c-ares.pdb $INSTALLDIR/lib/c-ares.pdb",
```

Comment:
Yeah, to the extent you can push in most of this that make sense and parameterize by path it'd be good. We can still add a postfix script for additional stuff as needed on each target.